src/ogr/featuredefn.jl

```diff
@@ -330,7 +330,7 @@ function directly, but use OGRLayer::DeleteGeomField() instead.
 This method should only be called while there are no OGRFeature objects in
 existence based on this OGRFeatureDefn.
 """
-function deletegeomdefn!(featuredefn::FeatureDefn, i::Integer)::FeatureDefn
+function deletegeomdefn!(featuredefn::Union{FeatureDefn, IFeatureDefnView}, i::Integer)
```

Comment:
I didn't think it through, but without this the GPKG workaround will fail.

Reply:
I might be convinced otherwise, but I doublechecked and don't think this is the way to go; https://github.com/yeesian/ArchGDAL.jl/pull/315/files#r943083917.